### PR TITLE
enable null blob connection for public blob storage

### PR DESCRIPTION
**Pull request by sebastus**
### What does this PR do?
* add test for null blob connection string, interpret that to be a public account
* change UI so that nothing appears in the SAS token expiration area if public account
* change auth to blob storage so an empty sas token is accepted
* fix the makefile target "run-plugins"
* test empty sas token with the gnuradio/iqengine test
* add test for existing sas token with daniel esteves container

### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you successfully ran tests with your changes locally? Including container validation.
* [x] Have you lint your code locally prior to submission?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] Have you labelled this PR correctly? E.G. major, milestone, breaking, breaking-change, minor, feature, enhancement, patch, fix, chore, refactor etc

### Changes to Core Features:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
   "private": true,
   "dependencies": {
     "@azure/storage-blob": "^12.14.0",
+    "@azure/identity": "^3.2.3",
```
Comment (Nepomuceno) [inline client/package.json:7]: Is this package needed?

**Comment by 777arc** [comment]
Looks great!

**Review by 777arc — APPROVED** [review]

File: client/src/pages/repo-browser/RepositoryTile.tsx [hunk 2]
```diff
@@ -15,13 +15,27 @@ export const RepositoryTile = (props) => {
   const [isDisabled, setIsDisabled] = useState(false);
   const [isWarning, setIsWarning] = useState(false);
   const [isError, setIsError] = useState(false);
+  const [isPublic, setIsPublic] = useState(false);
```
Comment (Nepomuceno) [inline client/src/pages/repo-browser/RepositoryTile.tsx:18]: There is no need for this it awill not be displayed is there is no warning or errors so we can jsut skyp the validation in case there the sas token is empty

File: client/src/pages/repo-browser/RepositoryTile.tsx [hunk 3]
```diff
@@ -15,13 +15,27 @@ export const RepositoryTile = (props) => {
...
   const [dayDifference, setDayDifference] = useState<number>();
   const [expires, setExpires] = useState();
   const [writeableBool, setWriteableBool] = useState<any>();
 
   useEffect(() => {
-    const tempExpires = sasToken.slice(sasToken.search('se')).split('&')[0].slice(3, 13); // YEAR-MONTH-DAY
-    const writeable = sasToken.slice(sasToken.search('sp')).split('&')[0].includes('w'); // boolean
+    let writeable = false;
+    if (sasToken == '') {
+      setIsPublic(true);
```
Comment (Nepomuceno) [inline client/src/pages/repo-browser/RepositoryTile.tsx:26]: you could just use if(!!sastoken) and avoid the else

File: client/package.json (hunk 1 above)
Comment (sebastus) [reply client/package.json:7]: good catch

File: client/src/pages/repo-browser/RepositoryTile.tsx (hunk 2 above)
Comment (sebastus) [reply client/src/pages/repo-browser/RepositoryTile.tsx:18]: Line 64 - 

        {!isError && !isWarning && (

I am getting the text "SAS Token Expiration: {expires}". I checked the history for the file and didn't see any changes to it, so I don't think it's an error. Adding !IsPublic removes it.

File: client/src/pages/repo-browser/RepositoryTile.tsx (hunk 3 above)
Comment (sebastus) [reply client/src/pages/repo-browser/RepositoryTile.tsx:26]: still need IsPublic flag
